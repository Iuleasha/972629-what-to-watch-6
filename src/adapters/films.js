export const filmAdapter = ({
  id,
  name = ``,
  poster_image: posterImage,
  preview_image: previewImage,
  background_image: backgroundImage,
  background_color: backgroundColor,
  description = ``,
  rating = 0,
  scores_count: scoresCount = 0,
  director = ``,
  starring: starring = [],
  run_time: runTime = 0,
  genre,
  released,
  is_favorite: isFavorite = false,
  video_link: videoLink,
  preview_video_link: previewVideoLink,
}) => {
  return {
    id,
    name,
    posterImage,
    previewImage,
    backgroundImage,
    backgroundColor,
    description,
    rating,
    scoresCount,
    director,
    starring,
    runTime,
    genre,
    released,
    isFavorite,
    videoLink,
    previewVideoLink,
  };
};

export const filmsAdapter = (films) => films.map(filmAdapter);
