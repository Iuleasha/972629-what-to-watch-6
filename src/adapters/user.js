export const userAdapter = ({
  id = ``,
  email = ``,
  name = ``,
  avatar_url: avatarUrl = ``,
}) => {
  return {
    id: id.toString(),
    email,
    name,
    avatarUrl,
  };
};
