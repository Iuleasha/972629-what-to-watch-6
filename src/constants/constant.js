export const HeaderMode = {
  MOVIE_CARD: `movie-card__head`,
  USER_PAGE: `user-page__head`,
};

export const PosterSize = {
  WIDTH: 218,
  HEIGHT: 327,
};

export const MovieCardButtonSize = {
  WIDTH: 19,
  HEIGHT: 20,
};

export const MOVIE_CARD_NAV_ITEMS = [`Overview`, `Details`, `Reviews`];

export const DEFAULT_GENRE = `All genres`;

export const AppRoute = {
  LOGIN: `/login`,
  LOSE: `/lose`,
  MY_LIST: `/mylist`,
  ROOT: `/`,
  FILMS: `/films`,
  FILM: `/films/:id`,
  PLAYER: `/player/:id`,
  REVIEW: `/films/:id/review`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const APIRoute = {
  FILMS: `/films`,
  FILM: `/film`,
  PROMO_FILM: `/films/promo`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`
};

export const ButtonIcon = {
  PLAY: `#play-s`,
  PAUSE: `#pause`,
  FULL_SCREEN: `#full-screen`,
  INLIST: `#in-list`,
  ADD: `#add`,
};

export const NameSpace = {
  DATA: `DATA`,
  COMMENTS: `COMMENTS`,
  USER: `USER`,
};

export const ActionType = {
  LOAD_SUCCESS: `load/loadSuccess`,
  LOAD_PROMO_FILM: `load/loadPromoFilm`,
  SET_GENRE: `filter/genres`,
  LOAD_FAVORITE: `load/favorite`,
  SWITCH_FAVORITE_STATUS: `load/switchFavoriteStatus`,
  REQUIRED_AUTHORIZATION: `user/requiredAuthorization`,
  REDIRECT_TO_ROUTE: `site/redirectToRoute`,
  SET_USER: `user/setUser`,
  ADD_REVIEW: `film/addReview`,
};

export const FILMS_PAGE_SIZE = 8;

export const AVATAR_SIZE = 63;

export const CommentLength = {
  MIN: 50,
  MAX: 400
};

export const STARS_LENGTH = 10;

export const RATING_DEFAULT_VALUE = 3;

export const RATING_STARS = [...Array(STARS_LENGTH).keys()].map((i) => i + 1);

export const Rating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};
