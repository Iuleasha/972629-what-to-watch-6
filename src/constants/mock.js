export const FILM_MOCK = {
  'id': 1,
  'name': `The Grand Budapest Hotel`,
  'poster_image': `img/the-grand-budapest-hotel-poster.jpg`,
  'preview_image': `img/the-grand-budapest-hotel.jpg`,
  'background_image': `img/the-grand-budapest-hotel-bg.jpg`,
  'background_color': `#ffffff`,
  'video_link': `https://some-link`,
  'preview_video_link': `https://some-link`,
  'description': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  'rating': 8.9,
  'scores_count': 240,
  'director': `Wes Andreson`,
  'starring': [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  'run_time': 99,
  'genre': `Comedy`,
  'released': 2014,
  'is_favorite': false,
};

export const COMMENT_MOCK = {
  'id': 1,
  'user': {
    'id': 4,
    'name': `Kate Muir`,
  },
  'rating': 8.9,
  'comment': `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  'date': `2019-05-08T14:13:56.569Z`,
};

export const AUTH_INFO_MOCK = {
  'id': 1,
  'email': `Oliver.conner@gmail.com`,
  'name': `Oliver.conner`,
  'avatar_url': `img/1.png`,
};
