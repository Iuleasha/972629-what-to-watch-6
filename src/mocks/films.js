export default [
  {
    'id': 1,
    'name': `The Grand Budapest Hotel`,
    'posterImage': `img/bg-the-grand-budapest-hotel.jpg`,
    'previewImage': `img/bg-the-grand-budapest-hotel.jpg`,
    'backgroundImage': `img/bg-the-grand-budapest-hotel.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'description': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 8.9,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 240,
    'director': `Wes Andreson`,
    'starring': [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    'runTime': 99,
    'genre': `Comedies`,
    'released': 2014,
    'isFavorite': false,
  },
  {
    'id': 2,
    'name': `Fantastic Beasts: The Crimes of Grindelwald`,
    'posterImage': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'previewImage': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'backgroundImage': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `Film follows Newt Scamander and Albus Dumbledore as they attempt to take down the dark wizard Gellert Grindelwald while facing new threats in a more divided wizarding world.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 7.6,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 180,
    'director': `David Yates`,
    'starring': [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`, `Alison Sudol`, `Ezra Miller`, `ZoÃ« Kravitz`],
    'runTime': 134,
    'genre': `Crime`,
    'released': 2018,
    'isFavorite': false,
  },
  {
    'id': 3,
    'name': `Bohemian Rhapsody`,
    'posterImage': `img/bohemian-rhapsody.jpg`,
    'previewImage': `img/bohemian-rhapsody.jpg`,
    'backgroundImage': `img/bohemian-rhapsody.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `is a 2018 biographical drama film about Freddie Mercury, the lead singer of the British rock band Queen.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 5.0,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 210,
    'director': `Graham King`,
    'starring': [`Peter Morgan`, `Rami Malek`, `Lucy Boynton`, `Gwilym Lee`, `Ben Hardy`],
    'runTime': 134,
    'genre': `Documentary`,
    'released': 2018,
    'isFavorite': false,
  },
  {
    'id': 4,
    'name': `Macbeth`,
    'posterImage': `img/macbeth.jpg`,
    'previewImage': `img/macbeth.jpg`,
    'backgroundImage': `img/macbeth.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `The story follows a Scottish general's rise to power after receiving a prophecy from a trio of witches that one day he will become King of Scotland.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 8.0,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 150,
    'director': `Justin Kurzel`,
    'starring': [`Michael Fassbender`, `Marion Cotillard`, `Paddy Considine`, `Sean Harris`],
    'runTime': 113,
    'genre': `Dramas`,
    'released': 2015,
    'isFavorite': false,
  },
  {
    'id': 5,
    'name': `Aviator`,
    'posterImage': `img/aviator.jpg`,
    'previewImage': `img/aviator.jpg`,
    'backgroundImage': `img/aviator.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `The film depicts the life of Howard Hughes, an aviation pioneer and director of the film Hell's Angels.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 7.7,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Martin Scorsese`,
    'starring': [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`, `John C. Reilly`],
    'runTime': 170,
    'genre': `Horror`,
    'released': 2004,
    'isFavorite': false,
  },
  {
    'id': 6,
    'name': `We need to talk about Kevin`,
    'posterImage': `img/we-need-to-talk-about-kevin.jpg`,
    'previewImage': `img/we-need-to-talk-about-kevin.jpg`,
    'backgroundImage': `img/we-need-to-talk-about-kevin.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `The gripping international bestseller about motherhood gone awry.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 8.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 120,
    'director': `Lynne Ramsay`,
    'starring': [`Tilda Swinton`, `John C. Reilly`, `Ezra Miller`],
    'runTime': 112,
    'genre': `Kids & Family`,
    'released': 2011,
    'isFavorite': false,
  },
  {
    'id': 7,
    'name': `What We Do in the Shadows`,
    'posterImage': `img/what-we-do-in-the-shadows.jpg`,
    'previewImage': `img/what-we-do-in-the-shadows.jpg`,
    'backgroundImage': `img/what-we-do-in-the-shadows.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `The gripping international bestseller about motherhood gone awry.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 6.3,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 170,
    'director': `Jemaine Clement`,
    'starring': [`Jemaine Clement`, `Taika Waititi`, `Jonathan Brugh`, `Cori Gonzalez-Macuer`],
    'runTime': 85,
    'genre': `Romance`,
    'released': 2014,
    'isFavorite': false,
  },
  {
    'id': 8,
    'name': `Revenant`,
    'posterImage': `img/revenant.jpg`,
    'previewImage': `img/revenant.jpg`,
    'backgroundImage': `img/revenant.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 9.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Alejandro G. IÃ±Ã¡rritu`,
    'starring': [`Leonardo DiCaprio`, `Tom Hardy`, `Domhnall Gleeson`, `Will Poulter`],
    'runTime': 156,
    'genre': `Sci-Fi`,
    'released': 2015,
    'isFavorite': false,
  },
  {
    'id': 9,
    'name': `Revenant`,
    'posterImage': `img/revenant.jpg`,
    'previewImage': `img/revenant.jpg`,
    'backgroundImage': `img/revenant.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 9.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Alejandro G. IÃ±Ã¡rritu`,
    'starring': [`Leonardo DiCaprio`, `Tom Hardy`, `Domhnall Gleeson`, `Will Poulter`],
    'runTime': 156,
    'genre': `Sci-Fi`,
    'released': 2015,
    'isFavorite': false,
  },
  {
    'id': 10,
    'name': `Revenant`,
    'posterImage': `img/what-we-do-in-the-shadows.jpg`,
    'previewImage': `img/revenant.jpg`,
    'backgroundImage': `img/revenant.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 9.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Alejandro G. IÃ±Ã¡rritu`,
    'starring': [`Leonardo DiCaprio`, `Tom Hardy`, `Domhnall Gleeson`, `Will Poulter`],
    'runTime': 156,
    'genre': `Sci-Fi`,
    'released': 2015,
    'isFavorite': false,
  },
  {
    'id': 11,
    'name': `Revenant`,
    'posterImage': `img/what-we-do-in-the-shadows.jpg`,
    'previewImage': `img/revenant.jpg`,
    'backgroundImage': `img/revenant.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 9.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Alejandro G. IÃ±Ã¡rritu`,
    'starring': [`Leonardo DiCaprio`, `Tom Hardy`, `Domhnall Gleeson`, `Will Poulter`],
    'runTime': 156,
    'genre': `Sci-Fi`,
    'released': 2015,
    'isFavorite': false,
  },
  {
    'id': 12,
    'name': `Revenant`,
    'posterImage': `img/what-we-do-in-the-shadows.jpg`,
    'previewImage': `img/revenant.jpg`,
    'backgroundImage': `img/revenant.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 9.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Alejandro G. IÃ±Ã¡rritu`,
    'starring': [`Leonardo DiCaprio`, `Tom Hardy`, `Domhnall Gleeson`, `Will Poulter`],
    'runTime': 156,
    'genre': `Sci-Fi`,
    'released': 2015,
    'isFavorite': false,
  },
  {
    'id': 13,
    'name': `Revenant`,
    'posterImage': `img/what-we-do-in-the-shadows.jpg`,
    'previewImage': `img/revenant.jpg`,
    'backgroundImage': `img/revenant.jpg`,
    'backgroundColor': `#ffffff`,
    'videoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'previewVideoLink': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'description': `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    'fullDescription': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder`,
    'rating': 9.1,
    'noteDescription': [`Bad`, `Normal`, `Good`, `Very good`, `Awesome`],
    'scoresCount': 230,
    'director': `Alejandro G. IÃ±Ã¡rritu`,
    'starring': [`Leonardo DiCaprio`, `Tom Hardy`, `Domhnall Gleeson`, `Will Poulter`],
    'runTime': 156,
    'genre': `Sci-Fi`,
    'released': 2015,
    'isFavorite': false,
  },
];
