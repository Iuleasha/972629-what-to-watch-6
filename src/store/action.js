import {createAction} from '@reduxjs/toolkit';
import {ActionType} from '../constants/constant';

const getPayload = (payload) => ({payload});

export const setGenre = createAction(ActionType.SET_GENRE, getPayload);

export const loadFilms = createAction(ActionType.LOAD_SUCCESS, getPayload);

export const loadPromoFilm = createAction(ActionType.LOAD_PROMO_FILM, getPayload);

export const requireAuthorization = createAction(ActionType.REQUIRED_AUTHORIZATION, getPayload);

export const redirectToRoute = createAction(ActionType.REDIRECT_TO_ROUTE, getPayload);

export const setUser = createAction(ActionType.SET_USER, getPayload);

export const addReview = createAction(ActionType.ADD_REVIEW, getPayload);

export const loadFavorite = createAction(ActionType.LOAD_FAVORITE, getPayload);

export const switchFavoriteStatus = createAction(ActionType.SWITCH_FAVORITE_STATUS, getPayload);
