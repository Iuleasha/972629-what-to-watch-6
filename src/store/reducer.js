import {AuthorizationStatus, DEFAULT_GENRE} from '../constant';
import {createGenreArray} from '../filter/filter';
import {ActionType} from './action';

const initialState = {
  genre: DEFAULT_GENRE,
  genres: [],
  films: [],
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isDataLoaded: false,
  comments: {},
  preview: null,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };

    case ActionType.LOAD_SUCCESS:
      return {
        ...state,
        genres: createGenreArray(action.payload),
        films: action.payload,
        isDataLoaded: true,
      };

    case ActionType.LOAD_PROMO_FILM:
      return {
        ...state,
        preview: action.payload,
      };

    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload,
      };

    case ActionType.SET_USER:
      return {
        ...state,
        user: action.payload,
      };

    case ActionType.ADD_REVIEW:
      const {comments} = state;

      return {
        ...state,
        comments: {...comments, [action.payload.id]: action.payload.comments},
      };

    case ActionType.LOAD_FAVORITE:
      return {
        ...state,
        favorite: action.payload,
      };

    case ActionType.SWITCH_FAVORITE_STATUS:
      if (state.preview && state.preview.id === action.payload.id) {
        state.preview = action.payload;
      }

      state.films = state.films.map((item) => item.id === action.payload.id ?
        action.payload : item
      );

      return {
        ...state,
        favorite: state.films.filter(({isFavorite}) => isFavorite),
      };
  }

  return state;
};


export {reducer};
