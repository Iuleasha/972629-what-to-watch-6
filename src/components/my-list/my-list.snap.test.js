import {configureStore} from '@reduxjs/toolkit';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import React from 'react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {filmsAdapter} from '../../adapters/films';
import {FILM_MOCK} from '../../constants/mock';
import {loadFavorite} from '../../store/action';
import reducer from '../../store/root-reducer';
import MyList from './my-list';

const store = configureStore({reducer});

store.dispatch((dispatch) => dispatch(loadFavorite(filmsAdapter([FILM_MOCK]))));

test(`Should MyList render correctly`, () => {
  const history = createMemoryHistory();

  const {container} = render(<Provider store={store}>
    <Router history={history}>
      <MyList/>
    </Router>
  </Provider>);

  expect(container).toMatchSnapshot();
});
