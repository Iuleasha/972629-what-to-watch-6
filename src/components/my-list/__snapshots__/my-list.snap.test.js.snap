// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyList render correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        class="page-title user-page__title"
      >
        My list
      </h1>
      <div
        class="user-block"
      >
        <a
          class="user-block__link"
          href="/login"
        >
          Sign in
        </a>
      </div>
    </header>
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <a
            class="small-movie-card__link"
            href="/films/1"
          >
            <div
              class="small-movie-card__image"
            >
              <img
                alt="The Grand Budapest Hotel"
                height="175"
                src="img/the-grand-budapest-hotel-poster.jpg"
                width="280"
              />
            </div>
            <h3
              class="small-movie-card__title"
            >
              <span>
                The Grand Budapest Hotel
              </span>
            </h3>
          </a>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 
          2021
           What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
